### PR TITLE
move AccountInfo to its own file

### PR DESCRIPTION
**Pull request by jeffwashington**
#### Problem
Refactoring of AccountInfo and AccountsIndex going on. Moving this to its own file allows me to hide struct members and move to accessors or traits.
#### Summary of Changes

Fixes #

### CODE REVIEW

File: runtime/src/accounts_db.rs [hunk 1]
```diff
@@ -284,22 +285,6 @@ impl GenerateIndexTimings {
...
-
-    /// needed to track shrink candidacy in bytes. Used to update the number
-    /// of alive bytes in an AppendVec as newer slots purge outdated entries
-    stored_size: usize,
-
-    /// lamports in the account used when squashing kept for optimization
-    /// purposes to remove accounts with zero balance.
-    lamports: u64,
-}
 impl IsCached for AccountInfo {
```
Comment (brooksprumo) [inline runtime/src/accounts_db.rs:288]: Should this be moved over to the new file too?

File: runtime/src/accounts_db.rs [hunk 2]
```diff
@@ -308,12 +293,6 @@ impl IsCached for AccountInfo {
 
 impl IndexValue for AccountInfo {}
```
Comment (brooksprumo) [inline runtime/src/accounts_db.rs:294]: Should this be moved over to the new file too?

**Review by brooksprumo — COMMENTED** [review]
Very much in favor of anything that makes accounts_db.rs smaller. Thank you!

File: runtime/src/accounts_db.rs (hunk 2 above)
Comment (jeffwashington) [reply runtime/src/accounts_db.rs:294]: I was thinking these trait impls make sense here and they are not part of the implementation. I think of them is 'here's how to apply these traits to this ignorant data structure'

**Review by brooksprumo — DISMISSED** [review]

**Pull request merged**